### PR TITLE
feat(kit): `InputDateRange` add full support of `TUI_DATE_FORMAT`

### LINKED ISSUE #954: 🚀 - `TUI_DATE_FORMAT` add full support of all `TuiDateMode` ('DMY' | 'MDY' | 'YMD')
### Which @taiga-ui/* package(s) are relevant/releated to the feature request?

kit

### Description

We have token [TUI_DATE_FORMAT](https://github.com/TinkoffCreditSystems/taiga-ui/blob/main/projects/cdk/date-time/date-format.ts).

It fully supports only `DMY` (dd.mm.yyyy).
We should add support of `MDY` (mm.dd.yyyy) and `YMD`(yyyy.mm.dd).

`InputDate`, `InputDateTime` and `InputDateRange` should be customizable via this token.

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #954

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: projects/cdk/date-time/day-range.ts [hunk 1]
```diff
@@ -37,16 +38,28 @@ export class TuiDayRange extends TuiMonthRange {
...
      * @return normalized day range object
      */
-    static normalizeParse(rangeString: string): TuiDayRange {
-        const leftDay = TuiDay.normalizeParse(rangeString.slice(0, DATE_FILLER_LENGTH));
+    static normalizeParse(
+        rangeString: string,
+        dateMode: string | TuiDateMode = 'DMY',
+    ): TuiDayRange {
+        // Temporary type guard to satisfy ts-overloading of this method (TODO rm in v3.0)
+        const isDateMode = (dateMode: string): dateMode is TuiDateMode =>
```
Comment (vladimirpotekhin) [inline projects/cdk/date-time/day-range.ts:55]: can be done as a separate function?

File: projects/demo/src/modules/components/input-date-range/input-date-range.template.html [hunk 2]
```diff
@@ -19,6 +19,28 @@
         >
             <tui-input-date-range-example-2></tui-input-date-range-example-2>
         </tui-doc-example>
+
+        <tui-doc-example
+            id="date-localization"
+            heading="Date localization"
+            [content]="example3"
+        >
+            <h3 i18n>DI-tokens for date format localization:</h3>
```
Comment (vladimirpotekhin) [inline projects/demo/src/modules/components/input-date-range/input-date-range.template.html:28]: nit: shouldn't we move the customization description from the concrete example to general description? Just like in dialog, input-number, swipe etc

**Review by vladimirpotekhin — APPROVED** [review]

**Review by splincode — APPROVED** [review]

File: projects/cdk/date-time/day-range.ts (hunk 1 above)
Comment (nsbarsukov) [reply projects/cdk/date-time/day-range.ts]: done

File: projects/demo/src/modules/components/input-date-range/input-date-range.template.html (hunk 2 above)
Comment (nsbarsukov) [reply projects/demo/src/modules/components/input-date-range/input-date-range.template.html]: done

**Pull request merged**